src/field/field_data.cxx

```diff
@@ -24,17 +27,17 @@ void FieldData::setBoundary(const string &name) {
   
   output_info << "Setting boundary for variable " << name << endl;
   /// Loop over the mesh boundary regions
-  for(const auto& reg : mesh->getBoundaries()) {
+  for(const auto& reg : getDataMesh()->getBoundaries()) {
```

Comment:
I think the constructor ensures that fielddatamesh can never be `nullptr` so it might be ok to just reference `fielddatamesh` here rather than asking `getDataMesh` to return it?

Reply:
I introduced this because of errors with the MMS/diffusion test: if a Field3D is declared in global scope, then it's constructor is called before the global `mesh` is allocated/initialized. Then fielddatamesh will be `nullptr` unless `getDataMesh()` is called to replace `nullptr` with the value of `mesh` once `mesh` has been initialized. So i think `getDataMesh` is necessary.

Reply:
PS is it better to have the check in `getDataMesh()` be `if (fielddatamesh != nullptr)` instead of `if (fielddatamesh)`?

Reply:
I'm ok with this, but stuff declared at global scope is a bit dubious